### PR TITLE
Migrate last legacy metadata payload: agent_checks

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

This PR is best review commit by commit.

This PR migrate the last legacy metadata payload to component and remove the legacy metadata system.

The last payload, `agent_checks`, is still part of the collector.

### Describe how to test/QA your changes

Test that the `agent_checks` metadata payload is still correctly send. Using a fake intake to print payload to `/intake` might be the easiest way.

### CODE REVIEW

File: comp/collector/collector/collectorimpl/collector.go [hunk 1]
```diff
@@ -86,20 +93,30 @@ func Module() fxutil.Module {
 
 func newProvides(deps dependencies) provides {
 	c := newCollector(deps)
+
+	var agentCheckMetadata metadata.Provider
+	if _, isSet := deps.MetricSerializer.Get(); isSet {
+		agentCheckMetadata = metadata.NewProvider(c.collectMetadata)
+	}
```
Comment (vickenty) [inline comp/collector/collector/collectorimpl/collector.go:100]: It seems that this component may end up providing nil Provider. Will all dependents be able to handle it?

Comment (hush-hush) [reply comp/collector/collector/collectorimpl/collector.go:100]: yes, we now filter groups on the user side rather than the provider side.

**Review by vickenty — APPROVED** [review]

File: comp/collector/collector/collectorimpl/agent_check_metadata.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+func (p *Payload) MarshalJSON() ([]byte, error) {
+	// use an alias to avoid infinite recursion while serializing
+	type PayloadAlias Payload
+
+	return json.Marshal((*PayloadAlias)(p))
+}
+
+// SplitPayload breaks the payload into times number of pieces
+func (p *Payload) SplitPayload(_ int) ([]marshaler.AbstractMarshaler, error) {
+	return nil, fmt.Errorf("AgentChecks Payload splitting is not implemented")
```
Comment (louis-cqrl) [inline comp/collector/collector/collectorimpl/agent_check_metadata.go:53]: Do we plan to implement it ? If yes maybe add a "yet" at the end of the sentence ?

Comment (hush-hush) [reply comp/collector/collector/collectorimpl/agent_check_metadata.go:53]: No, the `SplitPayload` is part of the `serializer` payload interface. It's called when a payload is too big for the intake. But there is no way to "split" a metadata payload into chunk and still be valid.

File: comp/collector/collector/collectorimpl/agent_check_metadata.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+	}
+
+	loaderErrors := collector.GetLoaderErrors()
+	for check, errs := range loaderErrors {
+		jsonErrs, err := json.Marshal(errs)
+		if err != nil {
+			log.Warnf("Error formatting loader error from check %s: %v", check, err)
+		}
+		status := []interface{}{
+			check, check, "initialization", "ERROR", string(jsonErrs),
```
Comment (louis-cqrl) [inline comp/collector/collector/collectorimpl/agent_check_metadata.go:100]: question maybe a bit stupid but why do we 6 elements in the list above and only 5 here ?

File: comp/collector/collector/collectorimpl/agent_check_metadata.go (hunk 2 above)
Comment (louis-cqrl) [reply comp/collector/collector/collectorimpl/agent_check_metadata.go:53]: okay thanks

**Review by louis-cqrl — APPROVED** [review]
LGTM for me but maybe ask someone who know components better than me to take a look in the team?

File: comp/collector/collector/collectorimpl/agent_check_metadata.go [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+			} else if len(s.LastWarnings) != 0 {
+				status = []interface{}{
+					s.CheckName, s.CheckName, s.CheckID, "WARNING", s.LastWarnings, "",
+				}
+			} else {
+				status = []interface{}{
+					s.CheckName, s.CheckName, s.CheckID, "OK", "", "",
+				}
+			}
+			if status != nil {
```
Comment (GustavoCaso) [inline comp/collector/collector/collectorimpl/agent_check_metadata.go:87]: If there a case in which status is nil? Mostly asking as there is the `else` branch which will always get reach of the previous two are not met. Maybe we can remove the `status != nil`

File: comp/collector/collector/collectorimpl/agent_check_metadata.go (hunk 3 above)
Comment (hush-hush) [reply comp/collector/collector/collectorimpl/agent_check_metadata.go:100]: No dumb questions ! This code has been moved without almost any change (not sure why GH can track the move).

Not sure why we have only 5 elem here.

File: comp/collector/collector/collectorimpl/agent_check_metadata.go (hunk 4 above)
Comment (hush-hush) [reply comp/collector/collector/collectorimpl/agent_check_metadata.go]: I think we can remove it. I move this code from the old check. Not sure why git can't track the change.

**Review by davidor — APPROVED** [review]
:+1: for container-integrations files

**Review by ogaca-dd — APPROVED** [review]

**Review by jose-manuel-almaza — APPROVED** [review]

**Review by AlexandreYang — APPROVED** [review]

**Review by GustavoCaso — APPROVED** [review]
👍 LGTM

**Pull request merged**